### ISSUE TITLE
shelly-1.6.3.3 test suite failure

### ISSUE THREAD

**Issue by peti** [yesodweb/Shelly.hs#108]
Citing from http://hydra.cryp.to/build/1066418/nixlog/1/raw:

~~~
Running 1 test suites...
Test suite shelly-testsuite: RUNNING...
bar

file with invalid encoding
  readBinary
  readfile
which
  gives full path to cabal
  recognizes cabal as a path executable
  cannot find missing exe
writefile
  creates and overwites a file
appendfile
  creates and appends a file
touchfile
  creates and updates a file
mv file
  to same dir
  to other dir
mv dir
  to dir does not exist: create the to dir
  to dir exists: creates a nested directory, full to path given
  to dir exists: creates a nested directory, partial to path given
rm file
  rm
  rm_r
  rm_f
rm_rf dir
  empty dir
  dir with file
rm symlink
  rm
  rm_f
  rm_rf
relativeTo
  relative to non-existent dir
  relative to existing dir
  abs path relative to existing dir
relative listing
  lists relative files
  finds relative files
find
  empty list for empty dir
  lists relative files
  lists absolute files
getting unset env variables
  get_env
  get_env_text
with SHELLY var set
  get_env
  get_env_text
failure set to stderr
testing
  writes a failure message to stderr
failure set to directory
testing
  writes a failure message to a .shelly directory
cp file
  cp to same dir
  cp to other dir
cp dir
  to dir does not exist: create the to dir
  to dir exists: creates a nested directory, full to path given
  to dir exists: creates a nested directory, partial to path given
  copies the same dir
basic actions
Hello!
  lifted sub
run
wibble
  simple command
*
  with escaping
dist LICENSE README.md Setup Setup.hs shelly.cabal src test
  without escaping
Selbstverständlich überraschend
  with binary handle mode
bash
wibble
  simple command
dist LICENSE README.md Setup Setup.hs shelly.cabal src test
  without escaping
Selbstverständlich überraschend
  with binary handle mode
  can detect failing commands in pipes
bar
  preserve pipe behaviour FAILED [1]

Failures:

  test/src/RunSpec.hs:49:
  1) bash preserve pipe behaviour
       expected: 0
        but got: 141

Randomized with seed 1630370220

Finished in 0.5302 seconds
51 examples, 1 failure
Test suite shelly-testsuite: FAIL
~~~

**Comment by peti** [comment]
This seems to be a duplicate of https://github.com/yesodweb/Shelly.hs/issues/106.

**Comment by peti** [comment]
This error occurs sporadically; maybe only once every 10 builds or so. I'm using NixOS Linux/x86_64 with GHC 7.10.2.

**Comment by gregwebs** [comment]
Thanks for reporting this. Closing as a dup of #106. If I don't come up with a solution today I will disable the test.
